src/core/utils.js

```diff
@@ -494,6 +495,30 @@ export function flatten(collector, item) {
 
 // --- DOM HELPERS -------------------------------
 
+// Takes an array and returns a string that separates each of its items with the proper commas and
+// "and". The second argument is a mapping function that can convert the items before they are
+// joined
+// Finally converts to hyperHTML while joining them
+
+export function htmlJoinAnd(array = [], mapper = item => item) {
+  const items = array.map(mapper);
+  switch (items.length) {
+    case 0:
+    case 1: // "x"
+      return items[0];
+    case 2: // x and y
+      return hyperHTML`${items[0]} and ${items[1]}`;
+    default: {
+      // x, y, and z
+      let joinedItems = ``;
+      for (let i = 0; i < array.length - 1; i++) {
+        joinedItems = `${joinedItems}, ${items[i]}`;
+      }
+      joinedItems = `${joinedItems} and ${items[length - 1]}`;
```

Comment:
The result of this loop is a bit broken by two ways.

* Consider we have `["x", "y", "z"]` in the array. `joinedItems` will first be `""`, and then `", x"`, `", x, y"`, and finally `", x, y and z"`, which is not what we want. The result should be: `"x, y, and z"`.
* This still does interpolation *before* we pass it into hyperHTML. We should do interpolation *with* hyperHTML so that it can process `HTMLElement`s properly, not stringifying it into `[object HTMLElement]`.

Tests are passing anyway, as it turns out there is no test for `conf.wgHTML`...

But still, this PR is now small enough to review and doesn't break any other things, which is a big 👍! 😁

Reply:
@saschanaz I have updated the PR. Please review it once more. 
Thanks for helping me so much